cmd/telemeter-server/main.go

```diff
@@ -428,14 +428,43 @@ func (o *Options) Run() error {
 					auth,
 				).ServeHTTP)
 
+			forwardClient := http.Client{
+				Timeout: 5 * time.Second,
+				Transport: telemeter_http.NewInstrumentedRoundTripper("forward",
+					&http.Transport{
+						MaxIdleConnsPerHost: 10,
+						IdleConnTimeout:     30 * time.Second,
+						DialContext:         (&net.Dialer{Timeout: 10 * time.Second}).DialContext,
+					},
+				),
+			}
+
+			if o.OIDCIssuer != "" {
```

Comment:
We already check for the existince of an OIDC issuer and set up an OIDC client and oauth client credentials on https://github.com/openshift/telemeter/blob/master/cmd/telemeter-server/main.go#L246. Can we simply reuse this to avoid code duplication?

Reply:
Yes, let me check how to make the first one not as scoped up there so we can more easily reuse it.